### ISSUE TITLE
Problem Publishing to Web Server

### ISSUE THREAD

**Issue by RGray1959** [bvcms/bvcms#45]
I am trying to assist my church in testing BVCMS for use in our Faith Formation program (initially) and potentially for use across all ministries.  Unfortunately, the situation is that I have to show them (my parish committee, 3 people) that the solution will work for their needs before they are going to go through an implementation and commit to a hosted solution and allocating any budget to the project.

Installed VS 2013 Professional, SQL Server 2012 Developer and IIS 7.0 on Server 2008 R2 (dual Xeon 3.0, 24Gb memory).  This was an 'old' server and software versions I have licenses to, and I would expect they should be usable with this application but tell me if there is a problem.  Downloaded the application and brought it up in VS per the instructions.  Had no problem with establishing the databases on SQL Server.  Opened it up in VS.

When starting in VS 2013 I receive the login just fine.  Reset the administrator password, and looked into the settings.  Changed a few (you can see in the attached images).  

![vs login works](https://cloud.githubusercontent.com/assets/10246960/6881438/1699fdfe-d52e-11e4-803a-0e0f270b95b1.png)

I am able to work with the software just fine, and see the changes saved in the database.  So all seems just fine.  

![vs application responds](https://cloud.githubusercontent.com/assets/10246960/6881439/34d3a32e-d52e-11e4-85f1-ecd3a51ee4d5.png)

All of this was done from within the VS environment after opening a project from a file location.  I created a new website under IIS7 server called “my-cgs.com”.  Then I published the site from VS to the server.  Had to work through a few settings changes (application pool bindings, .NET version, etc.) but it published successfully according to VS.

At first, from the website all I got was “The database does not exist…” so I worked with the connection string information, first on the VS side to make sure I wasn’t ‘breaking’ what was working, then publishing those changes.  Seemed to solve the problem of finding the database, without affecting my ability to operate the application within VS.

So now I receive this error from the published website:

![iis7 application doesnt respond](https://cloud.githubusercontent.com/assets/10246960/6881444/47ec28fa-d52e-11e4-9519-e111323edfe1.png)

I have tried any number of things (setting an initial default page, security/access checks on DB & file locations) but nothing seems to change this.  I am looking for suggestions on how to get the app that I see running in VS to publish and display in IIS7.  What should I try next?

**Comment by davidcarroll** [comment]
One of our developers will give you some suggestions sometime next week. If you would like to have our support staff give your church a full demo and a brief on the benefits of a hosting and support plan at a very reasonable fee (probably much less than the cost of your time and effort to host yourself) we would be happy to do that for you. Please contact info@touchpointsoftware.com. And of course if you are a developer, that would give you the best of both, ability to work on special reports/features for your church and take advantage of the efficiency of allowing us to host and support staff and users. The database will still be available to you as nightly backups you could download.

**Comment by RGray1959** [comment]
Thanks for your prompt response.  I will look forward to the suggestions when available.

That is ultimately my goal, to get to a full-blown hosted solution using all the features.  They are currently bouncing around looking at all kinds of options, without going through a real definition of goals and setting a project budget.  I am trying to lead them toward a good approach, but have to go carefully.  This all started when our new pastor said that he wanted "online registration" of our 900+ faith formation students ready for next fall's classes.  I have supported the MS Access database they have been using for about 10 years.  Beyond that, there has been very little guidance on where this project is going, but the new pastor is very interested in reaching out to our younger members through new media.  I have personally looked at 10-15 applications for and made recommendations, with your software being the one currently at the top of my list because of its open design and flexibility.  I can do them (and the project) a lot of good on the SQL Reporting side, as well as linking the database info into other processes (like their required donations reporting up to the Diocese) done with spreadsheets or mail merge, as that is my area of real strength with 30+ years of spreadsheet automation and spreadsheet/database integration.  I would be glad to contribute those types of things back to the project.

**Comment by drmohundro** [comment]
It's difficult to say exactly what is wrong here... I'll throw a few suggestions out to see if we can track down something:

1. In your `web.config` file, is your `host` setting set to the name of your church database (minus "CMS_")? It would look like `<add key="host" value="yourchurchname" />`.
2. When you run it from Visual Studio, it will run in IIS Express, but here you're running in full IIS. Can you make sure that your new site in IIS is running under an AppPool that is enabled for .NET 4.0? The default app pool is set up for only .NET 2.0.
3. Are there any errors in your Event Viewer? IIS may be running into other errors that are getting logged there, but the error page can't be displayed (and thus you're receiving a 404).
4. What do your bindings look like for this website in IIS? They'll need to be set up so that they match the URL you're using as well as the correct IP (or all IPs).

**Comment by RGray1959** [comment]
Thanks for your response.  Originally the web app was saying the DB’s were not set up when I knew they were, and doing #1 fixed that.  During the build/rebuild process VS 13 warned me about #2 and I changed my AppPool settings for this applicaton and that issue went away.  My site bindings are set to separate between the two sites that are running on that server, and before putting the app on the site I tested by putting a temporary “default.html” page on and checking that the different home pages showed on both sites (numbersense.net & my-cgs.com) so I don’t think it is that.

 

I looked into my Event viewer & the Server configuration manager and there is some kind of a “warning” about not being able to verify access to the directory on the hard drive where I am storing the web pages.  I will work to resolve that this evening and will send a screenshot of the “warning” if I can’t get it resolved.

 

Thanks,

Richard Gray
